[KB-281] [KB-282] [KB-283] Background: Define OLS regression, Ridge regression, LASSO and Elastic Net regression (#192)
chhoumann/thesis-chemcam: report_thesis/src/sections/background/linear_and_regularization_models/lasso.tex

\subsubsection{Least Absolute Shrinkage and Selection Operator (LASSO) Regression}
Having introduced ridge regression, we now turn our attention to the \gls{lasso} regression based on \citet{James2023AnIS}.

\gls{lasso} regression addresses the same issue of multicollinearity as ridge regression but does so by introducing a different regularization term to the \gls{olsr} objective function.

> While ridge regression uses the $L_2$ norm, \gls{lasso} uses the $L_1$ norm, which has the distinct effect of performing feature selection by shrinking some coefficients to exactly zero:

Reviewer: Perhaps a few words on how shrinking the coeffecients leads to feature selection. Just so we clearly demonstrate that we actually know how that works.

$$
L_1 = \sum_{j=1}^{p} |\beta_j|,
$$

where $\beta_j$ are the regression coefficients, and $p$ is the number of features.
The \gls{lasso} regression objective function is defined as:

$$

> \text{RSS} + \lambda L_1,

Reviewer: Same as ridge regression. Feels like there needs to be an objective function that equals this.

\subsubsection{Least Absolute Shrinkage and Selection Operator (LASSO) Regression}
A disadvantage of ridge regression is that it includes all features in the final model.

> While it does shrink the coefficients towards zero, it does not eliminate any.

Reviewer 2: Should we clarify that the `elimination` here refers to setting the weight for that feature to zero? Or is it fine to imply like this. Feature obv. doesn't get deleted.

Author: ```suggestion
While it does shrink the coefficients towards zero, it does not set any to exactly zero.
```

This may not impact prediction accuracy but makes interpretation harder, especially with many variables.

> For example, in datasets with numerous features, we might prefer a model that includes only the most important ones.

Reviewer 2: Generalize?
```suggestion
For example, in datasets with numerous features, one might prefer a model that includes only the most important ones.
```

> However, ridge regression includes all features, and while increasing the penalty reduces coefficient sizes, it does not exclude variables\cite{James2023AnIS}.

Reviewer 2: When taking this line + 3 into account, it does actually sound like the features get removed. So maybe clarify?
'Effectively excluded, as their coefficients are set to 0' or something.

Author: ```suggestion
However, ridge regression includes all features, and while increasing the penalty reduces coefficient sizes, it does not set any coefficients to zero\cite{James2023AnIS}.
```
